modal/_functions.py

```diff
@@ -1760,13 +1760,15 @@ async def from_id(function_call_id: str, client: Optional[_Client] = None) -> "_
         return fc
 
     @staticmethod
-    async def gather(*function_calls: "_FunctionCall[T]") -> typing.Sequence[T]:
```

Comment:
This is going to mean that the return type is "wrong" in the case of an exception right? Any idea how `asyncio.gather` handles this? Might need to `@overload`

Reply:
+1 - add an @overload which differentiates the return type based on a Literal[True] value for the return_exceptions argument, and otherwise only returns Sequence[`T`] - that should fix the existing type assertion